lib/delayed/worker.rb

```diff
@@ -116,7 +116,7 @@ def self.setup_lifecycle
     end
 
     def self.reload_app?
-      defined?(ActionDispatch::Reloader) && Rails.application.config.cache_classes == false
+      Rails.application.reloaders.any?(&:updated?) && Rails.application.config.cache_classes == false
```

Comment:
Given that the `Rails.application.config.cache_classes == false` check is constant (performance-wise) and `Rails.application.reloaders.any?(&:updated?)` is not, you should swap the order of these checks.